packages/financial-templates-lib/test/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -178,6 +179,87 @@ contract("CryptoWatchPriceFeed.js", function() {
     assert.equal((await cryptoWatchPriceFeed.getHistoricalPrice(1588376521)).toString(), toWei("1.5"));
   });
 
+  it("Basic TWAP price", async function() {
+    cryptoWatchPriceFeed = new CryptoWatchPriceFeed(
+      dummyLogger,
+      web3,
+      apiKey,
+      exchange,
+      pair,
+      lookback,
+      networker,
+      getTime,
+      minTimeBetweenUpdates,
+      false,
+      18, // Add arbitrary decimal conversion and prove this works.
+      undefined,
+      120
+    );
+
+    // Inject data.
+    networker.getJsonReturns = [...validResponses];
+
+    await cryptoWatchPriceFeed.update();
+
+    // ((12 * 32) + (13 * 60) + (14 * 28)) / 120
+    assert.equal(cryptoWatchPriceFeed.getCurrentPrice().toString(), "1296666666666666666");
+  });
+
+  it("Basic TWAP historical price", async function() {
+    cryptoWatchPriceFeed = new CryptoWatchPriceFeed(
+      dummyLogger,
+      web3,
+      apiKey,
+      exchange,
+      pair,
+      lookback,
+      networker,
+      getTime,
+      minTimeBetweenUpdates,
+      false,
+      18, // Add arbitrary decimal conversion and prove this works.
+      undefined,
+      120
+    );
+
+    // Inject data.
+    networker.getJsonReturns = [...validResponses];
+
+    await cryptoWatchPriceFeed.update();
+
+    // ((12 * 60) + (11 * 60)) / 120
+    assert.equal((await cryptoWatchPriceFeed.getHistoricalPrice(1588376460)).toString(), web3.utils.toWei("1.15"));
+  });
+
+  it("TWAP fails if period starts before data", async function() {
```

Comment:
How does TWAP perform if some data is missing inside the twap window? Think about the cases where data is missing in the middle, beginning, and end of window

Reply:
It uses the same backing algorithm as the uniswap feed, which is to say that it doesn't know anything about the expected data frequency. The CW feed, in general, is agnostic to how "filled" the data is. All it cares about is that there is data _before_ the earliest point you care about (which is what this test case is testing). Is there a test case that you think should be added on this point?

Reply:
Added tests for each of these cases!